metadata/src/test/java/org/apache/kafka/controller/PartitionChangeBuilderTest.java

```diff
@@ -1233,4 +1233,72 @@ public void testEligibleLeaderReplicas_ElectLastKnownLeaderShouldFail() {
         // No change to the partition.
         assertEquals(Optional.empty(), builder.build());
     }
+
+    @Test
```

Comment:
```suggestion
    @EnumSource(value = Election.class, names = {"PREFERRED", "UNCLEAN"})
    @ParameterizedTest
    public void testEligibleLeaderReplicas_NotEligibleLastKnownLeader(Election election) {
```

Please also change `.setElection(Election.PREFERRED)` to `.setElection(election)`

In this way, we don't need another similar test.